src/serve/routes.ts

```diff
@@ -1010,10 +1009,9 @@ if (isCheloniaDashboard) {
           // header to allow the service worker to function. Details:
           // https://w3c.github.io/ServiceWorker/#service-worker-allowed
           if (request.path.includes('assets/js/sw-')) {
-            console.debug('adding header: Service-Worker-Allowed /')
-            if (request.response instanceof Boom.Boom) {
-              request.response.output.headers['Service-Worker-Allowed'] = '/'
-            } else {
+            // If `response` is an error, then no SW is being served
+            if (!(request.response instanceof Boom.Boom)) {
+              console.debug('adding header: Service-Worker-Allowed /')
               request.response.header('Service-Worker-Allowed', '/')
```

Comment:
Based on the discussion on Slack, add the header only for non-error responses (which also is how it works on GI)